src/main/java/homer/model/outlets/Outlet.java

```diff
@@ -49,26 +43,15 @@ public Outlet(final double state, final double minValue,
     public Outlet(final Outlet outlet) {
         Objects.requireNonNull(outlet);
         this.state = outlet.getState();
-        this.minValue = outlet.getMinPower();
-        this.maxValue = outlet.getMaxPower();
-    }
-
-    //TODO: Remove?
-    private double getMaxPower() {
-        return this.maxValue;
-    }
-
-    private double getMinPower() {
-        return this.minValue;
     }
 
     /**
      * 
      * @return The device state parameters for eg. on/off, intensity.
      */
     @Override
-    public Double getState() {
-        return this.state;
+    public DeviceState getState() {
```

Comment:
what is the point of casting here? `OutletState` should implement `DeviceState` and be its own class. Check out my commits to see how i did it